src/core/index.js

```diff
@@ -275,6 +266,40 @@ util.extend( corefn, {
     return this._private.container;
   },
 
+  mount: function( container, rendererOptions ){
+    if( container == null ){ return; }
+
+    let cy = this;
+    let _p = cy._private;
+    let options = _p.options;
+    
+    let rOpts = rendererOptions ? rendererOptions : { name: 'canvas' };
+    options.renderer = rOpts;
+
+    if( !is.htmlElement( container ) && is.htmlElement( container[0] ) ){
+      container = container[0];
+    }
+  
+    cy.unmount();
+
+    _p.container = container;
+    _p.styleEnabled = true;
+
+    cy.initRenderer( rOpts );
+
+    cy.startAnimationLoop();
+
+    cy.style( options.style );    
+  },
+
+  unmount: function(){
+    let cy = this;
+
+    cy.stopAnimationLoop();
+
+    cy.destroyRenderer();
```

Comment:
Do you think unmount should create a null renderer?  That is what currently happens if you set ```headless: true``` in the core initialization.  Right now unmount makes cy._private.renderer null.

If we create a null renderer on unmount, it means we would be calling initRenderer twice in mount() but it would make unmount() consistent with setting headless to true.  

We could also change the core ```headless()``` function to read ```cy._private.renderer.name``` instead of reading ```cy._private.options.renderer.name``` so that ```headless()``` returns true once ```unmount()``` is called.

Reply:
It's usually better to have the `NullRenderer`.  For dev-facing API, we should use a `NullRenderer`.  But that means that `mount()` shouldn't re-use `unmount()`.  We don't want a `NullRenderer` unnecessarily when we're just about to create a `CanvasRenderer`.

Re. `headless()`, I don't know if that would work generally.  Our internal renderers may follow that pattern, but I don't think that's part of any API contract.  The renderer API is already a bit loosely defined, and I'd rather keep the API space as small as possible for it until it's formally defined for external renderer extensions.  

I'd rather check something like `this._private.renderer.isHeadless()` by adding that function to the `BaseRenderer` (false) and the `NullRenderer` (true) -- so that future renderer extensions don't have to worry about defining special functions or properties.